syntax = "proto3";

package auth;

service AuthService {
  rpc GetRedirectUri (Empty) returns (RedirectUriResponse);
  rpc ExchangeCodeForToken (ExchangeCodeRequest) returns (ExchangeCodeResponse);
  rpc GetUserIDFromJwtToken (GetUserIDFromJwtTokenRequest) returns (GetUserIDFromJwtTokenResponse);
}

message Empty {}

message RedirectUriResponse {
  string redirect_uri = 1;
}

message ExchangeCodeRequest {
  string code = 1;
  string state = 2;
}

message ExchangeCodeResponse {
  int32 user_id = 1;
  string jwt_token = 2;
  string name = 3;
  string mail = 4;
}

message GetUserIDFromJwtTokenRequest {
  string jwt_token = 1;
}

message GetUserIDFromJwtTokenResponse {
  int32 user_id = 1;
}
